File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistDao.java
package pl.revanmj.smspasswordnotifier.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface WhitelistDao {
    @Query("SELECT * FROM Senders ORDER BY name ASC")
    LiveData<List<WhitelistItem>> getAll();

    @Query("SELECT * FROM Senders WHERE sender_id == :id")
    WhitelistItem getById(int id);

    @Query("SELECT * FROM Senders WHERE name == :name")
    WhitelistItem getByName(String name);

    @Insert
    void insertAll(WhitelistItem... items);

    @Insert
    void insert(WhitelistItem item);

    @Delete
    void deleteAll(WhitelistItem... items);

    @Delete
    void delete(WhitelistItem item);
}

File: app/src/main/java/pl/revanmj/smspasswordnotifier/data/WhitelistDao.kt
package pl.revanmj.smspasswordnotifier.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface WhitelistDao {
    @get:Query("SELECT * FROM Senders ORDER BY name ASC")
    val all: LiveData<List<WhitelistItem>>

    @Query("SELECT * FROM Senders WHERE sender_id == :id")
    fun getById(id: Int): WhitelistItem

    @Query("SELECT * FROM Senders WHERE name == :name")
    fun getByName(name: String): WhitelistItem

    @Insert
    fun insertAll(vararg items: WhitelistItem)

    @Insert
    fun insert(item: WhitelistItem)

    @Update
    fun update(item: WhitelistItem)

    @Update
    fun updateAll(vararg items: WhitelistItem)

    @Delete
    fun deleteAll(vararg items: WhitelistItem)

    @Delete
    fun delete(item: WhitelistItem)
}
